llvm/lib/Transforms/IPO/GlobalOpt.cpp

```diff
@@ -1901,6 +1902,20 @@ static void makeAllConstantUsesInstructions(Constant *C) {
   }
 }
 
+static bool IsRepo = false;
```

Comment:
Is it safe to have this as a static variable? I would expect the state to be local.

Reply:
> 
> 
> Is it safe to have this as a static variable? I would expect the state to be local.
>

I think it is safe. If not, please let me know.

> I would expect the state to be local.
>
Did you mean that the 'IsRepo' will pass by the function parameter. If yes, the following function APIs need to be modified.

1) optimizeGlobalsInModule
2) processGlobal
3) OptimizeFunctions
4) processInternalGlobal
5) OptimizeGlobalVars
6) CommitValueTo
7) BatchCommitValueTo
8) EvaluateStaticConstructor

I am not sure which way is better. Please let me know what do you think.

Reply:
If LLVM coding standards allow the use of global variables for state, then what you've got is the simplest solution.

Reply:
Could the global at least be a static variable inside `calculateGlobalDigest()` (with the decision being made in there as well)? Renaming `calculateGlobalDigest()` to something like `calculateRepositoryDigestForGlobal()` would probably be a good idea to emphasise that the function is relevant only to the prepo.